protocol/lavasession/consumer_session_manager.go

```diff
@@ -115,6 +125,11 @@ func (csm *ConsumerSessionManager) UpdateAllProviders(epoch uint64, pairingList
 	// reset session related metrics
 	csm.consumerMetricsManager.ResetSessionRelatedMetrics()
 	csm.providerOptimizer.UpdateWeights(CalcWeightsByStake(pairingList))
+
+	// Update the stake map for metrics
+	stakeMapForMetrics := GetStakeMapFromPairingList(pairingList)
```

Comment:
no need for this function, just use the one in line 127 (CalcWeightsByStake) store it in a variable and use it (its a map[string]int64 instead of coin sdk just change the metric to accept it)

Reply:
Fixed in 893ec733406274e7ae4f2ab6655b4d1c5b9d13ac